File: fixtures/hub-cobol-clbs-mini/batch/reference_emit_cardpay.cs
// EXPECTED: 125.00
// Reference emit for CARDPAY — EVALUATE pay option + late IF.
using System;
public static class CardpayRef {
  public static void Main() {
    string option = "P";
    double bal = 1000.00;
    double pct = 0.1000;
    double minPay = 50.00;
    int daysLate = 45;
    double lateFee = 25.00;
    double pay = option switch {
      "F" => bal,
      "P" => Math.Round(bal * pct, 2, MidpointRounding.AwayFromZero),
      "M" => minPay,
      _ => 0.0,
    };
    double total = daysLate > 30
      ? Math.Round(pay + lateFee, 2, MidpointRounding.AwayFromZero)
      : pay;
    Console.WriteLine(total.ToString("F2"));
  }
}

File: fixtures/hub-cobol-clbs-mini/batch/reference_emit_cardpay.py
"""Reference emit for CARDPAY — EVALUATE F/P/M + late IF → 125.00."""
# EXPECTED: 125.00
OPTION = "P"
BAL = 1000.00
PCT = 0.1000
MIN_PAY = 50.00
DAYS_LATE = 45
LATE_FEE = 25.00
if OPTION == "F":
    pay = BAL
elif OPTION == "P":
    pay = round(BAL * PCT + 1e-12, 2)
elif OPTION == "M":
    pay = MIN_PAY
else:
    pay = 0.0
total = round(pay + LATE_FEE + 1e-12, 2) if DAYS_LATE > 30 else pay
print(f"{total:.2f}")
